sdk-core/src/main/java/care/data4life/sdk/network/util/ClientFactory.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2021 D4L data4life gGmbH / All rights reserved.
+ *
+ * D4L owns all legal rights, title and interest in and to the Software Development Kit ("SDK"),
+ * including any intellectual property rights that subsist in the SDK.
+ *
+ * The SDK and its documentation may be accessed and used for viewing/review purposes only.
+ * Any usage of the SDK for other purposes, including usage for the development of
+ * applications/third-party applications shall require the conclusion of a license agreement
+ * between you and D4L.
+ *
+ * If you are interested in licensing the SDK for your own applications/third-party
+ * applications and/or if you’d like to contribute to the development of the SDK, please
+ * contact D4L by email to help@data4life.care.
+ */
+
+package care.data4life.sdk.network.util
+
+import care.data4life.auth.AuthorizationContract
+import care.data4life.sdk.network.NetworkingContract
+import care.data4life.sdk.network.util.interceptor.BasicAuthorizationInterceptor
+import care.data4life.sdk.network.util.interceptor.LoggingInterceptor
+import care.data4life.sdk.network.util.interceptor.OAuthAuthorizationInterceptor
+import care.data4life.sdk.network.util.interceptor.RetryInterceptor
+import care.data4life.sdk.network.util.interceptor.StaticAuthorizationInterceptor
+import care.data4life.sdk.network.util.interceptor.VersionInterceptor
+import okhttp3.OkHttpClient
+import java.util.concurrent.TimeUnit
+
+object ClientFactory : NetworkingContract.ClientFactory {
+    private fun setPin(
+        builder: OkHttpClient.Builder,
+        environment: NetworkingContract.Environment,
+        platform: String,
+    ): OkHttpClient.Builder {
+        return builder
+            .certificatePinner(
+                CertificatePinnerFactory.getInstance(platform, environment)
+            )
+    }
+
+    private fun determineAuthMethod(
```

Comment:
Since this method is actually adding the right interceptor according to which method it is, this method is kinda doing 2 things. Might be worth it to make a method that returns the right Interceptor according to authMethod and then adding the Interceptor later outside of the function.

I saw thats the pattern of most `determineX` methods, so i'd say same comment applies to the other ones as well

Reply:
The AuthInterceptors are not interchangeable. With a StaticToken we need one Interceptor and without we are need two. Having that in mind, we would need to return a Pair here, where the 2nd item is nullable. And then check the Pair again if the 2n item is actually null.
This way we spare one compare and one null.

Reply:
Also this belongs into the ClientFactory PR.

Reply:
so maybe just renaming it would be enough, to something like `addAuthorizationInterceptor`, and inside maybe a helper like `hasStaticAccessToken` for the 2 branches?
Or maybe taking the if one level up directly?

Reply:
I'll make the renaming...`hasStaticAccessToken` would be then only an alias for `staticAccessToken is ByteArray`, since static access and OAuth/Basic are mutual exclusive.

Reply:
and this is what the method is determining - use static access or OAuth/Basic.